### PR TITLE
Fix handling of hrtime in `Duration.decode`

### PR DESCRIPTION
**Pull request by fubhy**

### CODE REVIEW

File: packages/effect/test/Duration.test.ts [hunk 1]
```diff
@@ -36,6 +36,8 @@ describe("Duration", () => {
 
     expect(Duration.decode([500, 123456789])).toEqual(Duration.nanos(500123456789n))
     expect(Duration.decode([-500, 123456789])).toEqual(Duration.zero)
+    expect(Duration.decode([Infinity, 0])).toEqual(Duration.infinity)
+    expect(Duration.decode([-Infinity, 0])).toEqual(Duration.zero)
```
Comment (gcanti) [inline packages/effect/test/Duration.test.ts:40]: I would add tests for `NaN`

**Review by gcanti — CHANGES_REQUESTED** [review]

File: packages/effect/test/Duration.test.ts (hunk 1 above)
Comment (gcanti) [reply packages/effect/test/Duration.test.ts:40]: ...and remove `expect` :-P

File: .changeset/good-pillows-leave.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"effect": patch
+---
+
+Fixed handling of hrtime in `Duration.decode`
```
Comment (gcanti) [inline .changeset/good-pillows-leave.md:5]:
```suggestion
Improve `Duration.decode` Handling of High-Resolution Time

- **Ensured Immutability**: Added the `readonly` modifier to `[seconds: number, nanos: number]` in `DurationInput` to prevent accidental modifications.
- **Better Edge Case Handling**: Now correctly processes special values like `-Infinity` and `NaN` when they appear in the tuple representation of duration.
```

File: packages/effect/test/Duration.test.ts [hunk 3]
```diff
@@ -33,6 +33,9 @@ describe("Duration", () => {
 
     deepStrictEqual(Duration.decode([500, 123456789]), Duration.nanos(500123456789n))
     deepStrictEqual(Duration.decode([-500, 123456789]), Duration.zero)
+    deepStrictEqual(Duration.decode([Infinity, 0]), Duration.infinity)
+    deepStrictEqual(Duration.decode([-Infinity, 0]), Duration.zero)
+    deepStrictEqual(Duration.decode([NaN, 0]), Duration.zero)
```
Comment (gcanti) [inline packages/effect/test/Duration.test.ts:39]:
```suggestion
    deepStrictEqual(Duration.decode([0, Infinity]), Duration.infinity)
    deepStrictEqual(Duration.decode([0, -Infinity]), Duration.zero)
    deepStrictEqual(Duration.decode([0, NaN]), Duration.zero)
```

File: packages/effect/test/Duration.test.ts [hunk 4]
```diff
@@ -69,6 +72,9 @@ describe("Duration", () => {
 
     assertSome(Duration.decodeUnknown([500, 123456789]), Duration.nanos(500123456789n))
     assertSome(Duration.decodeUnknown([-500, 123456789]), Duration.zero)
+    assertSome(Duration.decodeUnknown([Infinity, 0]), Duration.infinity)
+    assertSome(Duration.decodeUnknown([-Infinity, 0]), Duration.zero)
+    assertSome(Duration.decodeUnknown([NaN, 0]), Duration.zero)
```
Comment (gcanti) [inline packages/effect/test/Duration.test.ts:78]:
```suggestion
    assertSome(Duration.decodeUnknown([0, Infinity]), Duration.infinity)
    assertSome(Duration.decodeUnknown([0, -Infinity]), Duration.zero)
    assertSome(Duration.decodeUnknown([0, NaN]), Duration.zero)
```

**Review by gcanti — CHANGES_REQUESTED** [review]

**Review by gcanti — APPROVED** [review]
